### PR TITLE
Fix constructor regression

### PR DESCRIPTION
**Pull request by ichernev**
This was caused by https://github.com/moment/moment/pull/2368/files#diff-e792da0601bb2705000e86ef203f8e73R61

We should have had a test to cover for this case :)

Fix for #2512

### CODE REVIEW

File: src/lib/moment/constructor.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ var updateInProgress = false;
 // Moment prototype object
 export function Moment(config) {
     copyConfig(this, config);
-    this._d = new Date(config._d.getTime());
+    this._d = new Date(config._d != null ? config._d.getTime() : NaN);
```
Comment (mj1856) [inline src/lib/moment/constructor.js]: Should be `!==`, yes?

Comment (ichernev) [inline src/lib/moment/constructor.js]: No. `!= null` catches `null` and `undefined`. In our case its actually `undefined``.

Comment (mj1856) [inline src/lib/moment/constructor.js]: Ok.  Just making sure.  Thanks.

**Comment by mj1856** [comment]
LGTM

**Comment by wyantb** [comment]
My bad.  +1 for more tests.

**Pull request merged**